ttnn/cpp/ttnn/operations/moreh/moreh_linear_backward/device/moreh_linear_backward_multi_core_program_factory.cpp

```diff
@@ -192,35 +191,35 @@ operation::ProgramWithCallbacks moreh_bias_backward_multi_core_h(const Tensor &o
         tile_offset += num_cols_per_core;
     }
 
-    auto override_runtime_arguments_callback = [reader_kernel_id, writer_kernel_id, num_cores_to_be_used, num_cores_y](
-                                                   const void *operation,
-                                                   const Program &program,
-                                                   const std::vector<Tensor> &input_tensors,
-                                                   const std::vector<std::optional<const Tensor>> &,
-                                                   const std::vector<Tensor> &output_tensors) {
-        log_debug(LogOp, "{}:{} args_callback ", __func__, __LINE__);
-        const auto &output_grad = input_tensors.at(0);
-        const auto &bias_grad = output_tensors.at(0);
-
-        Buffer *src_buffer = output_grad.buffer();
-        Buffer *dst_buffer = bias_grad.buffer();
-
-        for (uint32_t i = 0; i < num_cores_to_be_used; ++i) {
-            CoreCoord core = {i / num_cores_y, i % num_cores_y};
-            {
-                auto &runtime_args = GetRuntimeArgs(program, reader_kernel_id, core);
-                runtime_args[0] = src_buffer->address();
-            }
-
-            {
-                auto &runtime_args = GetRuntimeArgs(program, writer_kernel_id, core);
-                runtime_args[0] = dst_buffer->address();
-            }
+    return {std::move(program), {reader_kernel_id, writer_kernel_id, num_cores_to_be_used, num_cores_y}};
+}
+
+void MorehBiasAddBackwardOperation::MultiCoreProgramFactory::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    auto& program = cached_program.program;
+    auto& reader_kernel_id = cached_program.shared_variables.unary_reader_kernel_id;
+    auto& writer_kernel_id = cached_program.shared_variables.unary_writer_kernel_id;
+    auto num_cores_to_be_used = cached_program.shared_variables.num_cores_to_be_used;
+    auto num_cores_y = cached_program.shared_variables.num_cores_y;
+
+    auto output_grad_buffer  = tensor_args.output_grad.buffer();
+    // auto bias_grad_buffer = tensor_args.bias_grad.value().buffer();
```

Comment:
Please delete unnecessary code.

Reply:
resolved in https://github.com/tenstorrent/tt-metal/pull/12812/commits/8dec1ff696bd9948f9366cbee8ec5733a58f2081